### PR TITLE
[TO_REVIEW] Handle edge cases for CAN

### PR DESCRIPTION
**Pull request by YanisLalou**
This pull request includes several updates to the `cdd_loss` function in `skada/deep/losses.py` and adds new test cases to ensure edge cases are handled correctly.

### Improvements to `cdd_loss` function:

* [`skada/deep/losses.py`](diffhunk://#diff-0774706cdd6571bdf61f65dbc2fe0fd5c9e6ae928b28459a523339e962180212R256-R257): Stacked `source_centroids` using `torch.stack` to ensure they are properly aggregated.
* [`skada/deep/losses.py`](diffhunk://#diff-0774706cdd6571bdf61f65dbc2fe0fd5c9e6ae928b28459a523339e962180212L282-R289): Added a small epsilon (`eps`) to the median pairwise distance calculation to avoid division by zero errors.
* [`skada/deep/losses.py`](diffhunk://#diff-0774706cdd6571bdf61f65dbc2fe0fd5c9e6ae928b28459a523339e962180212L298-R338): Added checks to ensure intra-class and inter-class domain discrepancy measures (`e1`, `e2`, `e3`) are set to zero if their corresponding masks sum to zero. This prevents potential errors from invalid operations.

### Enhancements to tests:

* [`skada/deep/tests/test_deep_divergence.py`](diffhunk://#diff-4a4c6ab899ff86f7bc4d7db095dfc066f1a8b843ac58284dbc9cb8d565b5584bR200-R210): Added a new test function `test_cdd_loss_edge_cases` to verify that `cdd_loss` handles edge cases correctly, such as when all source features are identical.

### CODE REVIEW

File: skada/deep/losses.py [hunk 1]
```diff
@@ -279,10 +282,12 @@ def cdd_loss(
     mask_t = valid_classes[cluster_labels_t]
     features_t = features_t[mask_t]
     cluster_labels_t = cluster_labels_t[mask_t]
-
     # Define sigmas
     if sigmas is None:
-        median_pairwise_distance = torch.median(torch.cdist(features_s, features_s))
+        eps = 1e-7
```
Comment (tgnassou) [inline skada/deep/losses.py:287]: eps as parameter

**Review by tgnassou — APPROVED** [review]
